File: JavaScript/linkedlist/MergeList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var mergeTwoLists = function(l1, l2) {
  let dummy = new ListNode(0);
  let curr = dummy;
  while (l1 !== null && l2 !== null) {
    if (l1.val > l2.val) {
      curr.next = l2;
      l2 = l2.next;
    } else {
      curr.next = l1;
      l1 = l1.next;
    }
    curr = curr.next;
  }
  while (l1 !== null) {
    curr.next = l1;
    l1 = l1.next;
    curr = curr.next;
  }
  while (l2 !== null) {
    curr.next = l2;
    l2 = l2.next;
    curr = curr.next;
  }
  return dummy.next;
};

File: Python/linkedlist/MergeList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        l3 = ListNode(0)
        i, j, cur = l1, l2, l3
        while i and j:
            if i.val < j.val:
                cur.next = i
                i = i.next
            else:
                cur.next = j
                j = j.next
            cur = cur.next
        while i:
            cur.next = i
            i = i.next
            cur = cur.next
        while j:
            cur.next = j
            j = j.next
            cur = cur.next
            
        return l3.next
                